sql/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -227,8 +227,21 @@ message Expression {
     }
 
     message Struct {
-      DataType struct_type = 1;
+      // (Deprecated) The type of the struct.
+      //
+      // This field is deprecated since Spark 4.1+ because using DataType as the type of a struct
+      // is ambiguous. This field should only be set if the data_type_struct field is not set.
+      // Use data_type_struct field instead.
+      DataType struct_type = 1 [deprecated = true];
+
+      // (Required) The literal values that make up the struct elements.
       repeated Literal elements = 2;
+
+      // The type of the struct.
+      //
+      // Whether data_type_struct.fields.data_type should be set depends on
+      // whether each field's type can be inferred from the elements field.
+      DataType.Struct data_type_struct = 3;
```

Comment:
@heyihong sorry for the late comments, what about introducing a new message for this purpose?
for example, `SimplifiedStruct` or `LightStruct`, my feeling is that the conversion in the server side become too complex

Reply:
@zhengruifeng FYI, I simplified the implementation in this PR: https://github.com/apache/spark/pull/52098
, but it's not necessary to introduce a new message.